src/server/supplements/har/harTracer.ts

```diff
@@ -68,7 +70,7 @@ export class HarTracer {
     this._pageEntries.set(page, pageEntry);
     this._log.pages.push(pageEntry);
     page.on(Page.Events.Request, (request: network.Request) => this._onRequest(page, request));
-    page.on(Page.Events.Response, (response: network.Response) => this._onResponse(page, response));
+    page.on(Page.Events.RequestFinished, (request: network.Request) => this._onRequestFinished(page, request));
```

Comment:
Not all requests finish, some of them stall, and others fail. I think we should update half of the fields on Response event, and another half on RequestFinished/RequestFailed. WDYT?

Reply:
yes, this is a correct observation. Somehow it is feels messy though. Reading the code it's not super clear what gets done when. I do think we have no other way, this is how HTTP / Browser networking is.

Testing this is tricky. We would need a local server for tests that does not respond or stalls some requests.

Reply:
I think you can just add a test with a failed request and see that some information is there. I do not think that extra testing complexity will be worth it in this particular case.